src/lib/supabase/library/library.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  mockGetAllLibraryItems,
+  mockSearchLibraryItemsByAvailability,
+  mockSearchLibraryItemsByCategory,
+  mockSearchLibraryItemsByName,
+  mockSearchLibraryItemsByTags,
+  mockSearchAllLibraryItems,
+} from './mockData';
+import {
+  LibraryItem,
+  Hardware,
+  Book,
+  Film,
+  Controller,
+  Accessory,
+} from '~/models/models';
+
+//Stub Cloud Functions for the Library Page
+
+const getAllLibraryItems = (): Promise<
+  ReadonlyArray<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+> => {
+  const response = new Promise<
+    Array<LibraryItem | Hardware | Book | Film | Controller | Accessory>
+  >(resolve => {
+    setTimeout(() => {
+      resolve(mockGetAllLibraryItems);
+    }, 500);
+  });
+
+  return response;
+};
```

Comment:
Then you'd only have to give this that `TLibraryItems` type instead of the longer list.  Do this for each function this this file to keep it tidy.
```suggestion
const getAllLibraryItems = (): Promise<ReadonlyArray<TLibraryItems>> => {
  const response = new Promise<Array<TLibraryItems>>(resolve => {
    setTimeout(() => {
      resolve(mockGetAllLibraryItems);
    }, 500);
  });

  return response;
};
```